include/nil/crypto3/zk/commitments/polynomial/kzg.hpp

```diff
@@ -652,14 +656,16 @@ namespace nil {
                     }
 
                     void update_transcript(std::size_t batch_ind, typename KZGScheme::transcript_type &transcript) {
-                        std::vector<std::uint8_t> byteblob(KZGScheme::scalar_blob_size);
                         // Push commitments to transcript
                         transcript(_commitments[batch_ind]);
 
-                        //Push evaluation points to transcript
+                        // Push evaluation points to transcript
                         for( std::size_t i = 0; i < this->_z.get_batch_size(batch_ind); i++){
-                            for( std::size_t j = 0; j < this->_z.get_poly_points_number(batch_ind, i); j++){
-                                KZGScheme::bincode::template field_element_to_bytes<std::vector<std::uint8_t>::iterator>(this->_z.get(batch_ind, i, j), byteblob.begin(), byteblob.end());
+                            for( std::size_t j = 0; j < this->_z.get_poly_points_number(batch_ind, i); j++  ) {
+                                nil::marshalling::status_type status;
```

Comment:
Same here, the value of _z must be marshalled once, as a single marshalling structure.